### ISSUE TITLE
BCLib 0.3.0 makes some items from BetterEnd 0.11.0 invisible when played with other mods

### ISSUE THREAD

**Issue by finallion** [paulevsGitch/BCLib#31]
Fabric version: 0.36.1

The latest version makes items invisible when played with other mods. I found the issue separately with Mo'Colors and Artificial Foliage (https://github.com/finallion/Artificial_Foliage-FABRIC-/issues/10).
When using the earlier version of BCLib and BetterEnd this doesn't happen. 

Logs: 
[2021-08-11-6.log](https://github.com/paulevsGitch/BCLib/files/6968630/2021-08-11-6.log)

The log with Artificial Foliage Mod is the same.

Probably something to do with [14:58:10] [Worker-Main-12/WARN]: Unable to load model: 'minecraft:block' referenced from: minecraft:item_frame#map=true: java.io.FileNotFoundException: minecraft:models/block.json

**Comment by paulevsGitch** [comment]
Previous version was compatible with Fabric API, and this one also should be... But looks like not. Probably Fabric API triggers cache before it is filled by BCLib models, but this is also strange as there is a double check that models are loaded. Probably I can make alternative model loading process for BCLib, so it will not be interrupted by other mods

**Comment by paulevsGitch** [comment]
Well, reverting big code part to old loading system fixes this issue (which means there is something wrong with my model loading method), but the old code part *probably* will be incompatible with CTM. I will try to find better solution later again

**Comment by paulevsGitch** [comment]
I finally fixed this. I replaced all injection into model cache with Fabric API injections. This automatically solve all issues with CTM, Artificial Foliage and other mods.

And model loading is still enhanced, as it runs parallel as I wanted before, it just uses its own model map now.

![2021-08-13_02 46 56](https://user-images.githubusercontent.com/17707380/129284055-d790274f-63a0-4916-9759-7729019167b9.png)
